### PR TITLE
Add authors back to old blog posts

### PR DESCRIPTION
**Pull request by VirginiaDooley**

### CODE REVIEW

File: democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    try:
+        models.User.objects.get(username=username)
+    except models.User.DoesNotExist:
+        return None
+
+
+# Post.authors.add(user)
+def get_authors_for_past_posts(apps, schema_editor):
+    with open("/blog-authors-2022-06-28.csv") as csvfile:
+        Post = apps.get_model("hermes", "Post").objects.all()
```
Comment (symroe) [inline democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py:24]: This is a way of getting the _model_ not the queryset. So you want `Post = apps.get_model("hermes", "Post")` and then pass that object in to the `get_post_by_slug` function.

File: democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        User = apps.get_model("hermes", "User").objects.all()
```
Comment (symroe) [inline democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py:25]: auth / User

File: democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+from django.conf import settings
+from django.db import migrations, models
+import csv
+
+# Get the blog object by slug (if it exists, again, because local)
+def get_post_by_slug(slug):
+    try:
+        models.Post.objects.get(slug=slug)
```
Comment (symroe) [inline democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py:8]: return here

File: democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    except models.Post.DoesNotExist:
+        return None
+
+
+# Get the user object by username
+def get_user_by_username(username):
+    try:
+        models.User.objects.get(username=username)
```
Comment (symroe) [inline democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py:16]: return here

**Comment by VirginiaDooley** [comment]
[Restore old blog authors](https://trello.com/c/qg75IkUZ/3040-restore-old-blog-authors)

File: democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        reader = csv.DictReader(csvfile)
+        # parse the csv file for slugs, authors
+        for row in reader:
+            username = row["username"]
+            slug = row["slug"]
+            blog_post = get_post_by_slug(Post, slug)
+            if blog_post:
+                author = get_user_by_username(User, username)
+                if author:
+                    blog_post.authors.add(author)
```
Comment (symroe) [inline democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py:35]: This should be `author` (no `s`)

**Review by symroe — COMMENTED** [review]
Looking good. Couple of small comments, and we should be good to go.

File: democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ("hermes", "0009_alter_post_author"),
+    ]
+
+    operations = [
+        migrations.RunPython(get_authors_for_past_posts),
```
[the remark this answers is not in the document]
Comment (VirginiaDooley) [reply democracy_club/apps/hermes/migrations/0010_add_past_post_authors.py]: I've made those changes and reverted the csv file back to the original.

**Pull request merged**